src/utils/oauth2_util.py

```diff
@@ -1,8 +1,47 @@
 from datetime import datetime, timedelta
 
+from src.constants.oauth2 import Roles, Scopes
+from src.constants.http import HttpMethods
+from src.lib.entities.secured_http_request_uri import SecuredHttpRequestUrl
+from src.lib.entities.endpoint_roles_scopes import EndpointRolesScopes
 import bcrypt
 import jwt
 
+from src.exceptions.exceptions import UnAuthorizedEndpoint
+
+ENDPOINT_ROLES_MAP = {
+    SecuredHttpRequestUrl(
+        path="/chefs", method=HttpMethods.GET.value
+    ): EndpointRolesScopes(
+        roles=[Roles.ADMINISTRATOR.value], scopes=[Scopes.READ_WRITE.value]
+    ),
+    SecuredHttpRequestUrl(
+        path="/chefs/<chef_id>", method=HttpMethods.GET.value
+    ): EndpointRolesScopes(
+        roles=[Roles.ADMINISTRATOR.value], scopes=[Scopes.READ_WRITE.value]
+    ),
+    SecuredHttpRequestUrl(
+        path="/chefs", method=HttpMethods.POST.value
+    ): EndpointRolesScopes(
+        roles=[Roles.ADMINISTRATOR.value], scopes=[Scopes.READ_WRITE.value]
```

Comment:
This scope should be only `write` since it is a post. Any endpoint that modify resources in the server should have write scopes, and endpoints that only reads information should have both read and write scopes

Reply:
done.